framework/web/CUrlManager.php

```diff
@@ -689,7 +695,10 @@ public function __construct($route,$pattern)
 		$this->append=$p!==$pattern;
 		$p=trim($p,'/');
 		$this->template=preg_replace('/<(\w+):?.*?>/','<$1>',$p);
-		$this->pattern='/^'.strtr($this->template,$tr).'\/';
+		$p=$this->template;
+		if(!$this->parsingOnly)
+			$p=preg_replace_callback('/(?<=^|>)[^<]+(?=<|$)/',array($this,'escapeRegexpSpecialChars'),$p);
```

Comment:
> Fixed the bug that CUrlRule treated special regex characters differently in createUrl and parseUrl

Because it *might* be instrumental to use regex in parsing only url rules. Problem was that createUrl and parseUrl behaved differently. Since using createUrl with parsing-only rules is impossible (so bug does not manifest there), and using regex wildcards in urlRules is a valid option, I used this conditional. It's reasonable when you think about it.